backend/middlewares/auth.ts

```diff
@@ -21,6 +30,81 @@ export const getAccessToken = (req: Request): string | null => {
   return null;
 };
 
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+const callResolver = async (
+  resolver: IMiddlewareFunction<any, ExpressContext>,
+  resolve: Resolve,
+  parent: any,
+  args: { [key: string]: any },
+  context: ExpressContext,
+  info: GraphQLResolveInfo,
+  // eslint-disable-next-line consistent-return
+) => {
+  if ("resolve" in resolver) {
+    return resolver.resolve?.(resolve, parent, args, context, info);
+  }
+  if (typeof resolver === "function") {
+    return resolver(resolve, parent, args, context, info);
+  }
+};
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
```

Comment:
Instead of this, why not just set the return type to an `IMiddlewareFunction<any, ExpressContext>`?